lib/class-wp-rest-widgets-controller.php

```diff
@@ -116,7 +92,7 @@ public function register_routes() {
 			),
 			'schema' => array( $this, 'get_public_item_schema' ),
 		) );
-		register_rest_route( $this->namespace, '/' . $this->rest_base .'/types/(?P<type>[\w-]+)', array(
+		register_rest_route( $this->namespace, '/' . $this->rest_base_types . '/(?P<type>[\w-]+)', array(
```

Comment:
Scratch what I said about using an OPTIONS request for schema data instead of using wp/v2/widget-types as the response body is not cacheable.  So I am for keeping the current endpoint structure in master, but either way it is fine.  If you are trying to future proof by having /:id be used instead for when widgets are turned into post data, that is fine.  However, that is not how widgets are stored right now and I think waiting on a change to core for that method to make sense would pretty much put the breaks on this endpoint when it is totally possible to have a fully functional endpoint with the current implementation of widgets in core.